File: js/42840.js
function solution1(answers) {
    let answer = [];
    var lena = answers.length;
    var p1 = [1,2,3,4,5];
    var p2 = [2,1,2,3,2,4,2,5];
    var p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    function max(a,b){
      if(a>b) return a;
      else return b;
    }
    let value = 0;
    let cnt = [0,0,0];
    for (var i=0; i<lena;i++){
      if(p1[i%5]===answers[i]) cnt[0]++;
      if(p2[i%8]===answers[i]) cnt[1]++;
      if(p3[i%10]===answers[i]) cnt[2]++;
    }
    value = max(max(cnt[0],cnt[1]),cnt[2]);
    var idx = 0;
    for (var i=0;i<3;i++){
      if(cnt[i]===value){
        answer[idx] = i+1;
        idx += 1;
      }
    }
    return answer;
}

function solution(answers){
  var answer = [];
  var p1 = [1,2,3,4,5];
  var p2 = [2,1,2,3,2,4,2,5];
  var p3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];

  var a1c = answers.filter((a,i)=> a === p1[i%5]).length;
  var a2c = answers.filter((a,i)=> a === p2[i%8]).length;
  var a3c = answers.filter((a,i)=> a === p3[i%10]).length;
  var max = Math.max(a1c,a2c,a3c);

  if (a1c===max){answer.push(1)};
  if (a2c===max){answer.push(2)};
  if (a3c===max){answer.push(3)};

  return answer
}

var asw = [1,2,3,4,5];
console.log(solution(asw));

File: resol/42840.py
def solution(answers):
    each_ans = {1: [1, 2, 3, 4, 5],
                2: [2, 1, 2, 3, 2, 4, 2, 5],
                3: [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]}
    order = [5, 8, 10]
    score = {1: 0, 2: 0, 3: 0}
    lena = len(answers)
    for i in range(lena):
        for j in range(3):
            if each_ans[j + 1][i % order[j]] == answers[i]:
                score[j + 1] += 1
    sortScore = sorted(score.items(), key=lambda x: (-x[1], x[0]))
    if sortScore[0][1] == sortScore[2][1]:
        return [1, 2, 3]
    elif sortScore[0][1] == sortScore[1][1]:
        return [sortScore[0][0], sortScore[1][0]]
    else:
        return [sortScore[0][0]]

def solution1(answers):
    answer = []
    each_ans = {1: [1, 2, 3, 4, 5],
                2: [2, 1, 2, 3, 2, 4, 2, 5],
                3: [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]}
    order = [5, 8, 10]
    score = [0,0,0]
    lena = len(answers)
    for i in range(lena):
        for j in range(3):
            if each_ans[j + 1][i % order[j]] == answers[i]:
                score[j + 1] += 1
    maxs = max(score)
    for i in range(len(score)):
        if score[i] == maxs:
            answer.append(i+1)
    return answer

a = [1, 3, 2, 4, 2]
print(solution(a))
